### ISSUE TITLE
Issue with non-English characters in filename

### ISSUE THREAD

**Issue by jcdick1** [jellyfin/jellyfin#4555]
**Describe the bug**
Playback fails on titles with non-English characters in title

**System (please complete the following information):**
 - OS: Ubuntu 20.04
 - Virtualization: XCP
 - Clients: All
 - Browser: Firefox
 - Jellyfin Version: 10.6.4
 - Playback: Doesn't
 - Installed Plugins: Reports, Playback Reporting
 - Reverse Proxy: Caddy
 - Base URL: none
 - Networking: NAT
 - Storage: NFS

**To Reproduce**

1. In my case, the movie "Amélie" - note the e with accent aigu
2. Scan into library
3. Look at characteristics of file
4. JF can't parse for codecs or streams
5. Playback fails

**Expected behavior**
Playback should function

**Logs**
[2020-11-22 22:13:59.137 +00:00] [INF] Profile: "Unknown Profile", Path: "/mnt/media/video/movies/amï¿½lie_(2001).mkv", isEligibleForDirectPlay: True, isEligibleForDirectStream: True

It can't recognize the aigu

From JF FFmpeg log:

file:/mnt/media/video/movies/amï¿½lie_(2001).mkv: No such file or directory

It can't recognize the file, I'm assuming due to the "special character"

**Screenshots**
From the Reports plugin, the codecs columns are blank.

![special_character_Reports_plugin](https://user-images.githubusercontent.com/49927327/99919048-9809af00-2ce0-11eb-8932-d14eac7fc230.png)

You can see that it doesn't know the codecs or audio streams.  Note the replacement character in the file name.

**Additional context**

**Comment by BaronGreenback** [comment]
Tested with 10.7 on windows. Looks like it's been fixed.

**Comment by yangmingli08** [comment]
I have encountered the same issue. When I tried <gdigrab> command to catch a chrome window with non-English characters such as ¥, or any other 中文页面, an I/O error will occur with the message "Can't find window "¥中文页面。 - Google Chrome", aborting." When the non-English characters are replaced by English ones, the errors will disappear.

**Comment by cvium** [comment]
@jcdick1 let us know if it persists

**Comment by hvalev** [comment]
I'm having a very similar problem. Jellyfin can't seem to parse umlauts or letters with accents. Below is part of the log from a library scan:
```
file:/data/Series/The Umbrella Academy (2019) [imdbid-tt1312171]/Season 02/The Umbrella Academy (2019) - S02E07 - �ga for �ga [WEBRip-1080p][10bit][EAC3 5.1][x265].mkv: No such file or directory
[17:09:10] [ERR] [22] MediaBrowser.Providers.TV.EpisodeMetadataService: Error in ffprobe
System.Exception: ffprobe failed - streams and format are both null.
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.GetMediaInfoInternal(String inputPath, String primaryPath, MediaProtocol protocol, Boolean extractChapters, String probeSizeArgument, Boolean isAudio, Nullable`1 videoType, Boolean forceEnableLogging, CancellationToken cancellationToken)
   at MediaBrowser.Providers.MediaInfo.FFProbeVideoInfo.ProbeVideo[T](T item, MetadataRefreshOptions options, CancellationToken cancellationToken)
   at MediaBrowser.Providers.Manager.MetadataService`2.RunCustomProvider(ICustomMetadataProvider`1 provider, TItemType item, String logName, MetadataRefreshOptions options, RefreshResult refreshResult, CancellationToken cancellationToken)
```

I suppose that jellyfin can't parse the original path with the name (in this case it should be Öga for öga) which is then passed to ffmpeg as an invalid path.

I'm running jellyfin in a docker container on a host OS with the following versions and characteristics:
Version: 10.7.7
Operating System: Linux
Architecture: Arm64

**Comment by maurirope** [comment]
I'm having the same problem on Jellyfin 10.10.3 on Ubuntu 24.04 VM

[2025-01-17 08:59:57.422 -03:00] [ERR] Error in "MusicArtistResolver" resolving "/storage/MEDIA/Music/Cursi"
System.AggregateException: One or more errors occurred. (Could not find a part of the path '/storage/MEDIA/Music/Cursi/[2004] Coraz�n de hotel'.)
 ---> System.IO.DirectoryNotFoundException: Could not find a part of the path '/storage/MEDIA/Music/Cursi/[2004] Coraz�n de hotel'.
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerableFactory.FileSystemInfos(String directory, String expression, EnumerationOptions options, Boolean isNormalized)
   at System.IO.DirectoryInfo.InternalEnumerateInfos(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at MediaBrowser.Controller.Providers.DirectoryService.<>c.<GetFileSystemEntries>b__5_0(String p, IFileSystem fileSystem)
   at Emby.Server.Implementations.Library.Resolvers.Audio.MusicArtistResolver.<>c__DisplayClass6_0.<Resolve>b__1(FileSystemMetadata fileSystemInfo, ParallelLoopState state)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass43_0`2.<PartitionerForEachWorker>b__1(IEnumerator& partitionState, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options, Boolean stopOnFirstFailure)
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.ThrowSingleCancellationExceptionOrOtherException(ICollection exceptions, CancellationToken cancelToken, Exception otherException)
   at System.Threading.Tasks.Parallel.PartitionerForEachWorker[TSource,TLocal](Partitioner`1 source, ParallelOptions parallelOptions, Action`1 simpleBody, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, Action`2 body)
   at Emby.Server.Implementations.Library.Resolvers.Audio.MusicArtistResolver.Resolve(ItemResolveArgs args)
   at Emby.Server.Implementations.Library.LibraryManager.Resolve(ItemResolveArgs args, IItemResolver resolver)

**Comment by hvalev** [comment]
I remember this. Is it a network mount? For my case I had an nfs mount where for some reason unicode characters were not supported in favor of some more restrictive encoding. It was resolved once I switched to a samba share

**Comment by maurirope** [comment]
Yes it is an NFS share. I do think it must be something related to that, as in my previous setup with SMB that wasn't happening indeed. I've been thinking about moving to SMB because this NFS thing does seem to perform clunkier.

However I there must be something to fix or configure for the proper behavior to work, because there's probably plenty of NFS implementations around there that I would think are working fine, so the root cause is relevant to Jellyfin as a headsup for others at least. What things is an NFS implementation on a commercial NAS doing different that I'm not doing in my Linux native NFS implementation?

**Comment by hvalev** [comment]
It has to do with the encodings that the nfs client on your linux supports. In my case I was using a jetson nano board some years ago where the nfs client itself didn't have the appropriate encodings. So if memory serves me right, I had to recompile the kernel with the encodings in a way that they would be included in the nfs client. It's been a while since then so I don't remember all the details. I've since moved to samba and never looked back.
